File: 27_Remove_Element/RemoveElement.py

"""
不是要把val remove掉
list 重新排過一遍 不加val
記憶體還在值不在
"""
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        ptr = 0 
        for num in nums:
            if num != val:
                nums[ptr] = num
                ptr += 1
        return ptr
        # new_nums = [num for num in nums if num != val]
        # print(new_nums)
        # return len(new_nums)
        #TC: O(n): n is length of nums
        #SC: O(1), 

File: DSA/not_yet_category/27_Remove_Element/RemoveElement.go
package main

import (
	"fmt"
)

func removeElement(nums []int, val int) int {
	ptr := 0
	for i:=0; i < len(nums); i++{
		if nums[i] != val {
			nums[ptr] = nums[i]
			ptr++
		}
	}
	return ptr
}
// 	if nums == nil {
// 		return 0
// 	}
// 	ptr := 0
// 	for i := 0; i < len(nums); i++{
// 		if nums[i] != val {
// 			if i != ptr {
// 				nums[i], nums[j] = nums[j], nums[i]
// 				ptr ++
// 			} else {
// 				ptr ++
// 			}
// 		}
// 	}
// 	return ptr
// }
